File: lib/hosts/gist.js
const fs = require('fs');
const request = require('request');

const url = 'https://api.github.com/gists';

exports.upload = (fileName, filePath) => {
  const fileData = fs.readFileSync(filePath);

  const form = {
    public: true,
    files: {}
  };

  form.files[fileName] = {
    content: fileData.toString()
  };

  request.post(
    {
      url,
      body: JSON.stringify(form),
      headers: {
        'User-Agent': 'tallpants'
      }
    },
    (err, response, body) => {
      if (err) {
        console.log(`Upload failed: ${err}`);
      } else {
        const responseData = JSON.parse(body);
        console.log(responseData.html_url);
      }
    }
  );
};

File: shareanything/hosts/gist.py
from requests import post


def upload(filename, filepath):
    with open(filepath, "r") as f:
        file_contents = f.read()

    payload = {
        "public": True,
        "files": {
            filename: {
                "content": file_contents
            }
        }
    }

    headers = {
        "User-Agent": "tallpants"
    }

    response = post("https://api.github.com/gists",
                    headers=headers, json=payload)
    try:
        return response.json()["html_url"]
    except Exception as e:
        print("Upload to Gist failed")
        print(e)


print(upload('test.py', './test.py'))
